app-modules/engagement/src/Filament/Pages/Inbox.php

```diff
@@ -92,6 +92,9 @@ public function table(Table $table): Table
                     ->badge(),
                 TextColumn::make('status')
                     ->badge(),
+                TextColumn::make('sender_type')
+                    ->label('Relation')
+                    ->formatStateUsing(fn (EngagementResponse $record) => ucwords($record->sender_type)),
```

Comment:
This is missing being sortable as well as this table is missing a filter being added as requested.

Please also add a test for the filter once it is done, I don't think a test file exists for the class yet so you will need to add one. But feel free to only add the test for your filter, other tests could be added in scope later.